types/lib/model.d.ts

```diff
@@ -2846,10 +2846,11 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
 
 export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = new () => Model<TModelAttributes, TCreationAttributes>;
 
+type RemoveInstanceProperties<T> = { [K in keyof T]: T[K] };
```

Comment:
I am not 100% sure why this change works - but it removes the instance properties from `typeof Model` specifically `_attributes` and `_creationAttributes`.

My suspicion is that because it maps over the keys of the type it changes how typescript is interpreting the type. In the `ModelCtor` case, the type is `typeof Model` which is derived from the value `class Model`. So TS is being smart and knowing that `Model` and `M` which extends Model are not the same type, as `Model` has `_attributes = {}`. There is something about the type map which stops TS doing any apparent nominal type checking. IDK - this is pure speculation and I could be very wrong.

Reply:
I just experimented with this and, although it does appear to solve this issue, it unfortunately breaks other typing tests:
![image](https://user-images.githubusercontent.com/12589522/139754051-e5433669-a640-4407-832b-6afc58f61d30.png)